### ISSUE TITLE
Melhorias no Artigo

### ISSUE THREAD

**Issue by fsantanna-uerj** [aiires/EDL#1]
- Exemplo de expressividade não está bom. Você poderia ter criado uma função de multiplicação em C e ter expressividade similar, ex. `mul(x,y)`.
- Favor estender discussão última seção, justificando as conclusões.

**Comment by matheusairesdev** [comment]
Concordo professor, mas em C não seria trivial realizar uma multiplicação de matrizes mxn por nxp. Com uma multiplicação 1xn por nx1 ambas se comportariam como um vetor, podendo então fazer a operação comutativa - o que não é permitido matematicamente. Para realizar esse tipo de operação em C, imagino que seria necessário criar uma struct incluindo a matriz, x e y, esses dois últimos dando a dimensão da matriz. Dessa forma, haveria uma forte reestruturação do código, pois a própria matriz será do tipo struct o que mudaria a forma de acessar os dados.

**Comment by fsantanna** [comment]
Todo esse código (struct, dimensões, etc.) poderiam estar escondidos em uma biblioteca.
Assim, o código da aplicação principal não precisa definir essas características.

**Comment by matheusairesdev** [comment]
Professor, achei esse exemplo abaixo onde eu crio uma lista em Scilab e posso colocar um "nome" junto com um vetor para representar as notas de um aluno:

L=list('Nome',[10 6]);

onde eu consigo acessar os elementos pelo indece:

//for (i = 1; i<=2; i++)
for i:1 2
  L(i);

em C, eu teria que criar uma struct e fazer uma lista encadeada para simular a lista em Scilab, o que reestruturaria o código:

struct ListaTurma{
        ListaTurma *t;
        char *nome;
        int notas[2];
} Turma;

Turma L;
L.nome = "Exemplo";
L.notas = {10,6}
L.t = null;
(a implementação será corrigida, é apenas um rascunho)

Posso usar como exemplo de expressividade?

**Comment by fsantanna** [comment]
Não é um bom exemplo de expressividade.
Tente ir na linha das matrizes e discutir a questão de gerenciamento de memória.
Ou entender melhor a parte de EDO.
